### PR TITLE
refactor: batchstore migration

### PR DESCRIPTION
**Pull request by istae**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md)
- [ ] My change requires a documentation update and I have done it
- [x] I have added tests to cover my changes.

### Screenshots (if appropriate):

---
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2825)

### CODE REVIEW

File: pkg/postage/batchstore/store.go [hunk 1]
```diff
@@ -288,6 +305,80 @@ func (s *store) Reset() error {
 	return nil
 }
 
+// migrate converts batchstore to the refactored version, removes unused indexes, computes radius, and
+// and calculates available capacity.
+func (s *store) migrate() error {
```
Comment (acud) [inline pkg/postage/batchstore/store.go:310]: not sure why this is needed.... i would recommend to just delete all batchstore keys as part of a statestore migration and force the batchstore to resync all chain data.

Comment (istae) [reply pkg/postage/batchstore/store.go:310]: fair enough, my thought process was that we could get away with not having to force users to a full resync, but it is the simplest and the least risky method.

**Review by mrekucci — COMMENTED** [review]
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2825)* status: all files reviewed, 1 unresolved discussion (waiting on @acud)

___
*[pkg/postage/batchstore/store.go, line 92 at r1](https://reviewable.io/reviews/ethersphere/bee/2825#-Mwb5yhq1fPy9FbJtCvg:-Mwb5yhs6kQXj-5jFECR:b-liimwu) ([raw file](https://github.com/ethersphere/bee/blob/94644d03183b7c2789546502c76562ea4b821afc/pkg/postage/batchstore/store.go#L92)):*
> ```Go
> 	err = st.Get(batchstoreVersion, &migrated)
> 	if err != nil {
> 		if err == storage.ErrNotFound {
> ```

I'd suggest using `errors.Is(err, storage.ErrNotFound)` here, since it'll also capture nested errors.

**Review by mrekucci — APPROVED** [review]
> Reviewed 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2825)* status: all files reviewed, 1 unresolved discussion (waiting on @acud)

**Pull request merged**